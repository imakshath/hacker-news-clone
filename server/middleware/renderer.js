import React from 'react'
import ReactDOMServer from 'react-dom/server'
import { Provider as ReduxProvider } from 'react-redux';
// import our main App component
import App from '../../src/App';

const path = require("path");
const fs = require("fs");

export default (store) => (req, res, next) => {
    // point to the html file created by CRA's build tool
    const filePath = path.resolve(__dirname, '..', '..', 'build', 'index.html');

    fs.readFile(filePath, 'utf8', (err, htmlData) => {
        if (err) {
            return res.status(404).end()
        }

        const html = ReactDOMServer.renderToString(
            <ReduxProvider store={store}>
                <App />
            </ReduxProvider>
        );

        // inject the rendered app into our html and send it
        return res.send(
            htmlData.replace(
                '<div id="root"></div>',
                `<div id="root">${html}</div>`
            )
        );
    });
}