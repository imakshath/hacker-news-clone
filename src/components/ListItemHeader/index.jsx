import React from 'react';
    
function ListItemHeader() {
    return (
        <div data-testid="list-item-header" className="list-item-header">
            <div className="item">Comments</div>
            <div className="item">Vote Count</div>
            <div className="item">Up Vote</div>
            <div className="item">News Detail</div>
        </div>
    )
};

export default React.memo(ListItemHeader);