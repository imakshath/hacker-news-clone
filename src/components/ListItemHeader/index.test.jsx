import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ListItemHeader from '.';

afterEach(cleanup);

describe('<ListItemHeader /> component', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = render(
            <ListItemHeader />
        );
    });

    test('should render without crash', () => {
        expect(wrapper.container).toBeDefined();
    });

    test('should get by testid', () => {
        expect(wrapper.getByTestId('list-item-header')).toBeDefined();
        expect(wrapper.getByText('Comments')).toBeDefined();
        expect(wrapper.getByText('Vote Count')).toBeDefined();
        expect(wrapper.getByText('Up Vote')).toBeDefined();
        expect(wrapper.getByText('News Detail')).toBeDefined();
    });

    test('should match snapshot', () => {
        const tree = renderer.create(<ListItemHeader />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})