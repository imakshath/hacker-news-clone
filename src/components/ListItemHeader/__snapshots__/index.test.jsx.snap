// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItemHeader /> component should match snapshot 1`] = `
<div
  className="list-item-header"
  data-testid="list-item-header"
>
  <div
    className="item"
  >
    Comments
  </div>
  <div
    className="item"
  >
    Vote Count
  </div>
  <div
    className="item"
  >
    Up Vote
  </div>
  <div
    className="item"
  >
    News Detail
  </div>
</div>
`;
